Feature/header profile dropdown menu

makc-anisimov opened the pull request:
DropDown menu при нажатии на иконку пользователя (при авторизации)

makc-anisimov committed:
feat(widgets): add profile menu to header

makc-anisimov committed:
feat(widgets): header menuProfile

add clickoutside function to close via custom hook

makc-anisimov committed:
refactor lint and husky changes

makc-anisimov committed:
Merge branch 'develop' of github.com:Pet-projects-CodePET/Frontend into feature/header-profile-dropdown-menu

makc-anisimov committed:
Merge branch 'develop' of github.com:Pet-projects-CodePET/Frontend into feature/header-profile-dropdown-menu

makc-anisimov committed:
Merge branch 'develop' of github.com:Pet-projects-CodePET/Frontend into feature/header-profile-dropdown-menu

makc-anisimov committed:
fix(widgets): menuProfile попавил хук clickOutside, адаптировал верстку

устарнил ошибки линтера, проверил верстку на разных разрешениях

makc-anisimov committed:
Merge branch 'develop' of github.com:Pet-projects-CodePET/Frontend into feature/header-profile-dropdown-menu

jsapro commented on src/widgets/header/ui/header.tsx:
```diff
@@ -49,13 +63,44 @@ export const Header = ({ isLoggedIn }: { isLoggedIn: boolean }) => {
-				</div>
+				{isLoggedIn && (
+					<div className={styles.header__menuProfile_container}>
+						<button
```
menuProfile лучше вынести в отдельный компонент, чтобы Header каждый раз не рендерился заново, когда изменяется menuProfile

jsapro commented on src/widgets/header/ui/header.tsx:
```diff
@@ -49,13 +63,44 @@ export const Header = ({ isLoggedIn }: { isLoggedIn: boolean }) => {
+							<User className={styles.header__linkProfile} />
+						</button>
+						<ul
+							className={`${styles.header__menuProfile} ${isMenuProfileOpen ? styles.header__menuProfile_opened : ''}`}
```
там где задаются стили по условию удобно использовать clsx, для улучшения читабельности

jsapro commented on src/widgets/header/ui/header.module.scss:
```diff
@@ -42,9 +44,90 @@
+		border-radius: 12px;
+		overflow: hidden;
+		border: 1px solid $--border-default;
+		box-shadow: 2px 2px 12px 0px rgba(62, 56, 56, 0.05);
```
для всех цветов лучше использовать константы

jsapro commented on src/widgets/header/ui/header.module.scss:
```diff
@@ -42,9 +44,90 @@
+		justify-content: center;
+		align-items: flex-start;
+		list-style: none;
+		padding: 8px 0;
```
при padding: 8px 0; сверху и снизу образовывается белая полоса
на мой взгляд это не очень красиво
лучше когда padding: 0;

jsapro commented on src/widgets/header/ui/header.tsx:
```diff
@@ -49,13 +63,44 @@ export const Header = ({ isLoggedIn }: { isLoggedIn: boolean }) => {
+									<Exit className={styles.header__menuProfileItem_icon} />
+									Выход
+								</Link>
+							</li>
```
тут много повторяющихся элементов, поэтому по идее можно сделать массив и через map() рендерить его

Liolikmolo reviewed (commented):
А меню не должно закрываться, когда снова кликнешь по иконке профиля?
Сейчас, когда кликаешь второй раз оно по новой перерисовывается просто

makc-anisimov committed:
Merge branch 'develop' of github.com:Pet-projects-CodePET/Frontend into feature/header-profile-dropdown-menu

makc-anisimov committed:
feat(entities): add profile menu

добавлено меню профиля в виде отдельного компонента. Доработана вся функциональность.

makc-anisimov committed:
fix(entities): fix styles differencies

Julia-Papina commented on src/entities/menu-profile/ui/menu-profile.tsx:
```diff
@@ -0,0 +1,71 @@
+				className={`${styles.menuProfile__list} ${isMenuProfileOpen ? styles.menuProfile__list_opened : ''}`}
+				ref={menuProfileRef}>
+				<li className={styles.menuProfile__item}>
+					<Link href="'/profile" className={styles.menuProfile__link}>
+						<User className={styles.menuProfile__item_icon} />
+						Личный кабинет
+					</Link>
```
Опечатка в href, лишняя кавычка

makc-anisimov committed:
fix(features): fix syntax error: menu link
